agent/grpc/client_test.go

```diff
@@ -240,3 +243,41 @@ func registerWithGRPC(t *testing.T, b *resolver.ServerResolverBuilder) {
 		resolver.Deregister(b.Authority())
 	})
 }
+
+func TestRecoverMiddleware(t *testing.T) {
```

Comment:
This test looks great, but I think it should probably be moved to `handler_test.go` since it is mostly testing that the "handler" is being setup correctly.

I think this name would match our naming convention a bit better and make it easier to identify which behaviour is being tested:

```suggestion
func TestHandler_PanicRecoveryInterceptor(t *testing.T) {
```